mon_school/mon_school/doctype/lms_sketch_like/lms_sketch_like.py

```diff
@@ -0,0 +1,50 @@
+# Copyright (c) 2021, FOSS United and contributors
+# For license information, please see license.txt
+
+import frappe
+from frappe.model.document import Document
+
+
+class LMSSketchLike(Document):
+    def after_insert(self):
+        sketch = self.get_sketch()
+        sketch.update_metrics()
+
+    def after_delete(self):
+        sketch = self.get_sketch()
+        sketch.update_metrics()
+
+    def get_sketch(self):
+        return frappe.get_doc("LMS Sketch", self.sketch)
+
+
+def on_doctype_update():
+    # only one like for each sketch/user
+    frappe.db.add_unique("LMS Sketch Like", fields=["sketch", "user"])
+
+
+@frappe.whitelist()
+def toggle_sketch_like(sketch_name, action):
+    user = frappe.session.user
+
```

Comment:
You should not allow this when `frappe.session.user` is `"Guest"`.

Reply:
Nice catch. Thanks!